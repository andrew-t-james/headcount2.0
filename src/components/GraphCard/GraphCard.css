.graph-card {
  max-height: 35rem;
  padding: 2rem;
  width: 40%;
}


@media screen and (max-width: 1345px) {
  .graph-card {
    width: 100%;
  }
}