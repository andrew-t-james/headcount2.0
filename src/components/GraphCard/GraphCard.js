import React from 'react';
import PropTypes from 'prop-types';
import { Bar } from 'react-chartjs-2';

import './GraphCard.css';

const GraphCard = ({ location, stats }) => {
  const setColorsForValues = Object.values(stats)
    .map(val => val > 0.5 ? 'rgba(39, 174, 96, 0.4)' : 'rgba(235, 77, 75, 0.4)');
  const graphData = {
    labels: Object.keys(stats),
    datasets: [
      {
        label: location,
        backgroundColor: setColorsForValues,
        borderColor: setColorsForValues,
        borderWidth: 1,
        hoverBackgroundColor: setColorsForValues,
        hoverBorderColor: setColorsForValues,
        data: Object.values(stats)
      }
    ]
  };

  return (
    <div className="graph-card">
      <h2>{location}</h2>
      <Bar
        data={graphData}
        options={{
          legend: { display: false },
          responsive: true,
          maintainAspectRatio: false
        }}
      />
    </div>
  );
};

GraphCard.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object.isRequired
};

export default GraphCard;
