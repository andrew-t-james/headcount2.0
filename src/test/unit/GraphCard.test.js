import React from 'react';
import { shallow } from 'enzyme';
import GraphCard from '../../components/GraphCard/GraphCard';

describe('GraphCard unit test suite', () => {
  let wrapper;
  const mockDistrict = {
    location: 'COLORADO',
    stats: {
      2005: 0.353
    }
  };

  beforeEach(() => wrapper = shallow(<GraphCard {...mockDistrict}/>));

  afterEach(() => wrapper.unmount());

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
