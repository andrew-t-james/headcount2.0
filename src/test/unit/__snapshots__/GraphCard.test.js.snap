// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphCard unit test suite should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GraphCard
    location="COLORADO"
    stats={
        Object {
            "2005": 0.353,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          COLORADO
</h2>,
        <Bar
          data={
                    Object {
                              "datasets": Array [
                                Object {
                                  "backgroundColor": Array [
                                    "rgba(235, 77, 75, 0.4)",
                                  ],
                                  "borderColor": Array [
                                    "rgba(235, 77, 75, 0.4)",
                                  ],
                                  "borderWidth": 1,
                                  "data": Array [
                                    0.353,
                                  ],
                                  "hoverBackgroundColor": Array [
                                    "rgba(235, 77, 75, 0.4)",
                                  ],
                                  "hoverBorderColor": Array [
                                    "rgba(235, 77, 75, 0.4)",
                                  ],
                                  "label": "COLORADO",
                                },
                              ],
                              "labels": Array [
                                "2005",
                              ],
                            }
          }
          options={
                    Object {
                              "legend": Object {
                                "display": false,
                              },
                              "maintainAspectRatio": false,
                              "responsive": true,
                            }
          }
/>,
      ],
      "className": "graph-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "datasets": Array [
              Object {
                "backgroundColor": Array [
                  "rgba(235, 77, 75, 0.4)",
                ],
                "borderColor": Array [
                  "rgba(235, 77, 75, 0.4)",
                ],
                "borderWidth": 1,
                "data": Array [
                  0.353,
                ],
                "hoverBackgroundColor": Array [
                  "rgba(235, 77, 75, 0.4)",
                ],
                "hoverBorderColor": Array [
                  "rgba(235, 77, 75, 0.4)",
                ],
                "label": "COLORADO",
              },
            ],
            "labels": Array [
              "2005",
            ],
          },
          "options": Object {
            "legend": Object {
              "display": false,
            },
            "maintainAspectRatio": false,
            "responsive": true,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            COLORADO
</h2>,
          <Bar
            data={
                        Object {
                                    "datasets": Array [
                                      Object {
                                        "backgroundColor": Array [
                                          "rgba(235, 77, 75, 0.4)",
                                        ],
                                        "borderColor": Array [
                                          "rgba(235, 77, 75, 0.4)",
                                        ],
                                        "borderWidth": 1,
                                        "data": Array [
                                          0.353,
                                        ],
                                        "hoverBackgroundColor": Array [
                                          "rgba(235, 77, 75, 0.4)",
                                        ],
                                        "hoverBorderColor": Array [
                                          "rgba(235, 77, 75, 0.4)",
                                        ],
                                        "label": "COLORADO",
                                      },
                                    ],
                                    "labels": Array [
                                      "2005",
                                    ],
                                  }
            }
            options={
                        Object {
                                    "legend": Object {
                                      "display": false,
                                    },
                                    "maintainAspectRatio": false,
                                    "responsive": true,
                                  }
            }
/>,
        ],
        "className": "graph-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "rgba(235, 77, 75, 0.4)",
                  ],
                  "borderColor": Array [
                    "rgba(235, 77, 75, 0.4)",
                  ],
                  "borderWidth": 1,
                  "data": Array [
                    0.353,
                  ],
                  "hoverBackgroundColor": Array [
                    "rgba(235, 77, 75, 0.4)",
                  ],
                  "hoverBorderColor": Array [
                    "rgba(235, 77, 75, 0.4)",
                  ],
                  "label": "COLORADO",
                },
              ],
              "labels": Array [
                "2005",
              ],
            },
            "options": Object {
              "legend": Object {
                "display": false,
              },
              "maintainAspectRatio": false,
              "responsive": true,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
